lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1322,6 +1322,22 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
     return DbgTran->createDebugDeclarePlaceholder(cast<DbgDeclareInst>(II), BB);
   case Intrinsic::dbg_value:
     return DbgTran->createDebugValuePlaceholder(cast<DbgValueInst>(II), BB);
+  case Intrinsic::annotation: {
+    SPIRVType *Ty = transType(II->getType());
+
+    GetElementPtrInst *GEP = dyn_cast<GetElementPtrInst>(
+                                 II->getArgOperand(1));
```

Comment:
Is it possible to use getConstantStringInfo() to get AnnotationString from the operand.

Reply:
Personally, I would also prefer it done through `getConstantStringInfo()`. However, the original approach is consistent with handling of `var_annotation` and `ptr_annotation`. I believe refactoring all three had better be done separately

Reply:
Please add a TODO comment as a reminder to do re-factoring and possible using of `getConstantStringInfo()`